homeassistant/components/vesync/__init__.py

```diff
@@ -1 +1,133 @@
-"""The vesync component."""
+"""Etekcity VeSync integration."""
+import logging
+import voluptuous as vol
+from pyvesync import VeSync
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD,
+                                 CONF_TIME_ZONE)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.config_entries import SOURCE_IMPORT
+from .common import (async_process_devices, CONF_SWITCHES)
+from .config_flow import configured_instances
+from .const import (VS_DISPATCHERS, VS_DISCOVERY,
+                    SERVICE_UPDATE_DEVS, CONF_MANAGER)
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'vesync'
+
+DEFAULT_SCAN_INTERVAL = 36000
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_TIME_ZONE): cv.time_zone,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the VeSync component."""
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][CONF_SWITCHES] = []
+
+    conf = config.get('vesync')
+
+    if conf is None:
+        return True
+
+    if isinstance(conf, dict) and not configured_instances(hass):
```

Comment:
We don't need to check for a dict. That's already validated by the config schema.